main.tf

```
@@ -76,9 +76,9 @@ resource "aws_backup_plan" "default" {
       }
 
       dynamic "copy_action" {
-        for_each = lookup(rule.value, "destination_vault_arn", null) != null ? ["true"] : []
+        for_each = toset(lookup(rule.value, "destination_vault_arns", []))
         content {
-          destination_vault_arn = lookup(rule.value, "destination_vault_arn", null)
+          destination_vault_arn = copy_action.key
```

Shouldnt this be value?

```suggestion
          destination_vault_arn = copy_action.value
```

---

[key](https://www.terraform.io/language/expressions/dynamic-blocks#key) is the map key or list element index for the current element. If the for_each expression produces a set value then key is identical to value and should not be used.

Is probably better convention to use value but they are both identical for a set. I will change to value